src/backend/access/appendonly/aosegfiles.c

```diff
@@ -1680,7 +1680,7 @@ get_ao_distribution(PG_FUNCTION_ARGS)
 	funcctx = SRF_PERCALL_SETUP();
 
 	query_block = (QueryInfo *) funcctx->user_fctx;
-	if (query_block->index < query_block->rows)
+	if (query_block != NULL && query_block->index < query_block->rows)
 	{
 		/*
 		 * Get heaptuple from SPI, then deform it, and reform it using our
```

Comment:
Pointer SPI_tuptable cannot be NULL here, because it is allocated before at first [call](https://github.com/arenadata/gpdb/pull/753/files#diff-0c258aeffb2b9d76ae39361be9ee3daa35123e0359dd67dc8fa4be7685245b05R1634-R1656).